packages/concerto-core/src/introspect/decorator.js

```diff
@@ -106,6 +111,42 @@ class Decorator extends ModelElement {
     getArguments() {
         return this.arguments;
     }
+
+    /**
+     * Returns true if this class is the definition of a decorator.
+     *
+     * @return {boolean} true if the class is a decorator
+     */
+    isDecorator() {
+        return true;
+    }
+
+    /**
+     * Resolves the type of the decorator argument
+     * @param {Object} argument - the decorator argument // Idt we have argument as any defined type
+     * @return {ClassDeclaration} the type of the decorator argument // similar to [this](https://github.com/accordproject/concerto/blob/0d7f108d4961f87782436dd689639ce0f35e97f8/packages/concerto-core/lib/introspect/classdeclaration.js#L155) method
+     */
+    _resolveArguemntType(argument) {
+
+        let classDecl = null;
+
+        const modelFile = this.parent.getModelFile();
+
+        if (modelFile.isImportedType(argument.type.name)) {
```

Comment:
I think just `modelFile.getType()` should be able to give you classDeclacarations that are imported or present within the namespace.
https://github.com/accordproject/concerto/blob/3dddd1ec5c41be47d1e2f3070de5d979afa8e23f/packages/concerto-core/lib/introspect/modelfile.js#L409